src/fourc_webviewer/read_geometry_from_file.py

```diff
@@ -397,70 +452,104 @@ def check_for_geometry_files(fourc_yaml: FourCInput) -> list:
         self._fourc_yaml = FourCInput.from_4C_yaml(input_file_path=fourc_yaml_file)
 
-        # create vtu file based on the geometry type
+        # set path for the vtu file to be created based on the geometry type
         self._vtu_file_path = str(Path(temp_dir) / f"{Path(fourc_yaml_file).stem}.vtu")
-        if self.geom_type == "exo":
+
+        # check for the geometry type
+        if self.geom_type == "legacy":
+            # convert yaml file to vtu file and return the path to the vtu file
             try:
-                # read exodus mesh
-                self._mesh_file = (
-                    Path(fourc_yaml_file).parent
-                    / self._fourc_yaml["STRUCTURE GEOMETRY"]["FILE"]
+                self._dis = read(str(fourc_yaml_file))
+                self.convert_dis_to_vtu()
+            except Exception as exc:  # if file conversion not successful
+                # log unsuccessful conversion
+                print(f"ERROR: conversion to vtu was not successful")
+                print(
+                    exc
+                )  # currently, we throw the lnmmeshio conversion error as terminal output
+                self._vtu_file_path = ""
+        elif self.geom_type in ["exo", "vtu"]:
+            try:
+                # read mesh: for the first rendering, we take the relative path with respect to the yaml file; for subsequent renderings, we will account for the absolute path
+                self._mesh_file = Path(
+                    get_geometry_file(fourc_yaml=self._fourc_yaml)[0]
                 )
+                if first_render:
+                    self._mesh_file = Path(fourc_yaml_file).parent / self._mesh_file
+                else:
+                    self._mesh_file = self._mesh_file.resolve()
                 if not self._mesh_file.exists():
                     raise Exception(
                         f"The mesh file {self._mesh_file} does not exist for the fourc yaml file {fourc_yaml_file}"
                     )
-                self._mesh_exo = read_exodus(
-                    filename=self._mesh_file,
-                    use_set_names=False,
-                )
-                self._mesh_exo = switch_node_order(mesh_exo=self._mesh_exo)  # -> not
-                # yet, this will mess up the applied boundary
-                # conditions, because 4C uses another order format
+
+                # read and postprocess mesh
+                if self.geom_type == "exo":
+                    self._mesh = read_exodus(
+                        filename=self._mesh_file,
+                        use_set_names=False,
+                    )
+                    self._mesh = postprocess_exo_mesh(mesh=self._mesh)
+                elif self.geom_type == "vtu":
+                    self._mesh = read_mesh(
+                        filename=str(self._mesh_file.resolve()),
+                    )
+                    self._mesh = postprocess_vtu_mesh(mesh=self._mesh)
```

Comment:
I think the functionality should be in a separate function whose only job is to return the desired mesh from a given file. This way, this method would be completely independent of the mesh reading, which makes this easier to extend (and test). 

```suggestion
def read_geometry(mesh_file):
                # read and postprocess mesh
                if mesh_file.suffix in EXODUS_FILE_SUFFIXES:
                    mesh = read_exodus(
                        filename=mesh_file,
                        use_set_names=False,
                    )
                    return postprocess_exo_mesh(mesh=mesh)
                elif mesh_file.suffix in VTU_FILE_SUFFIXES:
                    mesh = read_mesh(
                        filename=str(mesh_file.resolve()),
                    )
                   return postprocess_vtu_mesh(mesh=mesh)
                else:
                    raise FileTypeError("Unsupported file format.")
```